compiler/shared_ast/type.ml

```diff
@@ -90,55 +75,183 @@ let rec compare ty1 ty2 =
   | _, TArray _ -> 1
   | TDefault _, _ -> -1
   | _, TDefault _ -> 1
-  | TClosureEnv, _ -> -1
-  | _, TClosureEnv -> 1
+  | TVar _, _ -> -1
+  | _, TVar _ -> 1
+  | TAny _, _ -> -1
+  | _, TAny _ -> 1
+(* | TClosureEnv, _ -> -1
```

Comment:
Why leaving this commented?

Reply:
Same convention as in `Expr`: for the sake of uniformity, I prefer to keep it here, but uncommenting it would trigger a `the match case is unused` warning.